app/src/main/java/de/trbnb/apptemplate/main/MainViewModel.kt

```diff
@@ -3,22 +3,20 @@ package de.trbnb.apptemplate.main
 import android.content.Context
 import android.databinding.Bindable
 import de.trbnb.apptemplate.R
-import de.trbnb.apptemplate.app.App
 import de.trbnb.apptemplate.second.SecondActivity
 import de.trbnb.databindingcommands.command.RuleCommand
 import de.trbnb.databindingcommands.command.SimpleCommand
 import de.trbnb.mvvmbase.BaseViewModel
 import de.trbnb.mvvmbase.bindableproperty.afterSet
 import de.trbnb.mvvmbase.bindableproperty.bindable
 import org.jetbrains.anko.startActivity
-import javax.inject.Inject
 
-class MainViewModel : BaseViewModel() {
+class MainViewModel(
+        // FIXME this leaks the context and prevents deviceless viewmodel tests
```

Comment:
Should this be removed? Context is also injected by Koin later on.

Reply:
Android Studio warns that this will leak the context. On advantage of a viewmodel is that it survives the activity, e.g. when the device rotates and the activity is killed and recreated. So if the viewmodel holds a reference to the old activity/view, it can't be garbage collected.
In this case it should be fine however, as it is the application context (injected by Koin), which lives as long as the app is running, not the activity context.

The main disadvantage is, that the viewmodel now references Android platform specific code and is not plain java anymore. So in order to test the viewmodel we can't just use the jvm and run tests instantly anywhere, but we have to instantiate an android device (emulator or real device) to run the tests on. 
There are solutions to this (see [Mock Android dependencies](https://developer.android.com/training/testing/unit-testing/local-unit-tests#mocking-dependencies)). Easiest would probably be to specify the viewmodel's dependencies through interfaces, e.g. a `ResourceProvider` for strings with a plain java mock-implementation for tests and an implementation for Android that uses the context.

If you want I can implement this solution. Sorry for the wall of text :grimacing: